src/metabase/driver.clj

```diff
@@ -602,3 +602,18 @@
   {:added "0.39.0" :arglists '([driver db-details])}
   dispatch-on-uninitialized-driver
   :hierarchy #'hierarchy)
+
+(defmulti normalize-db-details
+  "Normalizes db-details for the given driver. This is to handle migrations that are too difficult to perform via
+  regular Liquibase queries. This multimethod will be called from a `:post-select` handler within the database model.
+  The full `database` model object is passed as the 2nd parameter, and the multimethod implementation is expected to
+  update the value for `:details`. The default implementation is essentially `identity` (i.e returns `db-details`
+  unchanged)."
+  {:added "0.40.0" :arglists '([driver db-details])}
```

Comment:
0.41.0?

Reply:
> The full `database` model object is passed as the 2nd parameter

The arglist has `db-details` as the second parameter

Reply:
Fixed both.